sdk/iot/Azure.Iot.Hub.Service/src/API Design.md

```diff
@@ -45,79 +45,126 @@ APIs for managing device identities, device twins, and querying devices
+    /// </summary>
+    /// <param name="devices">The pairs of devices their twins that will be created. For fields such as deviceId
+    /// where device and twin have a definition, the device value will override the twin value.</param>
+    /// <param name="cancellationToken">The cancellation token.</param>
+    /// <returns>The result of the bulk operation.</returns>
+    public virtual async Task<Response<BulkRegistryOperationResult>> CreateIdentitiesWithTwinAsync(IDictionary<DeviceIdentity, TwinData> devices, CancellationToken cancellationToken = default)
+
+    /// <summary>
+    /// Create multiple devices with an initial twin. A maximum of 100 creations can be done per call, and each creation must have a unique device identity. For larger scale operations, consider using IoT Hub jobs (https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-identity-registry#import-and-export-device-identities).
     /// </summary>
     /// <param name="devices">The pairs of devices their twins that will be created. For fields such as deviceId
     /// where device and twin have a definition, the device value will override the twin value.</param>
     /// <param name="cancellationToken">The cancellation token.</param>
     /// <returns>The result of the bulk operation.</returns>
-    public async Task<Response<BulkRegistryOperationResult>> CreateIdentitiesWithTwinAsync(IDictionary<DeviceIdentity, TwinData> devices, CancellationToken cancellationToken = default)
+    public virtual Response<BulkRegistryOperationResult> CreateIdentitiesWithTwin(IDictionary<DeviceIdentity, TwinData> devices, CancellationToken cancellationToken = default)
 
     /// <summary>
-    /// Create multiple device identities. A maximum of 100 creations can be done per call, and each device identity must be unique. For larger scale operations, consider using IoT Hub jobs (https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-identity-registry#import-and-export-device-identities).
+    /// Create multiple devices. A maximum of 100 creations can be done per call, and each device identity must be unique. For larger scale operations, consider using IoT Hub jobs (https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-identity-registry#import-and-export-device-identities).
     /// </summary>
     /// <param name="deviceIdentities">The devices to create.</param>
     /// <param name="cancellationToken">The cancellation token.</param>
     /// <returns>The result of the bulk operation.</returns>
     public virtual async Task<Response<BulkRegistryOperationResult>> CreateIdentitiesAsync(IEnumerable<DeviceIdentity> deviceIdentities, CancellationToken cancellationToken = default)
```

Comment:
For this, can you check with SDK team about guidance on the type for input - ISet or IEnumerable or something else? I believe we only want to allow unique identities.

Reply:
The annoying thing here is that a user can create two device identities with the same device id, and the set wouldn't catch that

Reply:
But even still, that is a good question

Reply:
Same comment for everything that has IEnumerable<DeviceIdentity> deviceIdentities as a parameter.

Reply:
Leave it as it as but when you send an email about the IDictionary, just ask about this as well. That way we can represent our design choices well to the service team.